st2common/st2common/services/workflows.py

```diff
@@ -1561,3 +1564,38 @@ def identify_orphaned_workflows():
             continue
 
     return orphaned
+
+
+def add_system_info_to_action_context(func):
```

Comment:
Per my other comment, I think we should refactor this 1 or 2 separate functions and not use a decorator.